src/energy_systems/heat_sources/geothermal_probes.jl

```diff
@@ -172,8 +180,10 @@ mutable struct GeothermalProbes <: Component
 end
 
 function initialise!(unit::GeothermalProbes, sim_params::Dict{String,Any})
-    set_storage_transfer!(unit.input_interfaces[unit.m_heat_in],
-                          unload_storages(unit.controller, unit.m_heat_in))
+    if unit.regeneration
+        set_storage_transfer!(unit.input_interfaces[unit.m_heat_in],
```

Comment:
Both the `do_storage_transfer` flag on interfaces and the `unload_storages` function of Controller default to true. The purpose of setting the flag in `initialise!` is to set them to false if a corresponding control setting exists and it set to false. Is this on purpose, overriding the default only if regeneration mode is active? Is there a case where a user might want to disable storage unloading but not use regeneration mode?

Edit: I think I answered myself. If regeneration mode is disabled, no input energy is ever used, thus the storage unloading flag are moot.

Reply:
The actual problem that makes adding the flag necessary is that the input interface is not connected to another component when regeneration is false, so the interface is not initialised and therefore is "nothing". set_storage_transfer!() requires a variable of type Interface as the first argument, not of type Nothing, so an errer occurs.